tarogarden/seedling.go

```diff
@@ -124,6 +129,47 @@ func (c Seedling) validateFields() error {
 	return nil
 }
 
+// validateGroupKey attempts to validate that the non-zero group key provided
+// with a seedling is owned by the daemon and can be used with this seedling.
+func (c Seedling) validateGroupKey(groups []*asset.AssetGroup) (
+	*asset.AssetGroup, error) {
+
+	// Scan for a group that matches our seedling group key.
+	var seedlingGroup *asset.AssetGroup
+	for i, group := range groups {
+		if bytes.Equal(group.GroupPubKey.SerializeCompressed(),
```

Comment:
Somehow it feels wrong to compare serialized public keys that might be zero keys. Can't we just have the pubkey nil-able in both places and then add a helper function similar to `EqualKeyDescriptors()` in `asset.go`?